pdfbox/src/main/java/org/apache/pdfbox/pdmodel/PDResources.java

```diff
@@ -95,6 +96,24 @@ public PDResources(COSDictionary resourceDictionary, ResourceCache resourceCache
         }
         resources = resourceDictionary;
         cache = resourceCache;
+        directFontCache = new HashMap<COSName,SoftReference<PDFont>>();
+    }
+
+    /**
+     * Constructor for reading.
+     *
+     * @param resourceDictionary The cos dictionary for this resource.
+     * @param resourceCache The document's resource cache, may be null.
+     */
```

Comment:
please add a good javadoc line here

Reply:
While adding javadoc I realised that mutability of cache cannot be deduced from method signature by reader - someone could try to use immutable map as parameter, so new type is introduced that is mutable by design - PDFontCache.
Javadoc added also

Reply:
I don't like the idea to introduce a new type just to avoid that people might follow the bad idea to pass an immutable map for a cache. IMHO if the do so, they have to deal with an exception.

Reply:
Removed introduced type